Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -333,16 +361,31 @@ public async Task Submit(SubmitSubmissionServiceModel model)
                                  !participant.Contest.IsDeleted &&
                                  problem.ShowResults;
 
-        await this.submissionsData.Add(newSubmission);
-        await this.submissionsData.SaveChanges();
+        var submissionType = problem.SubmissionTypesInProblems
+            .First(st => st.SubmissionTypeId == model.SubmissionTypeId)
+            .SubmissionType;
+
+        if (submissionType.ExecutionStrategyType != ExecutionStrategyType.NotFound &&
```

Comment:
Are you sure that we should send the submission to the distributor if the `ExecutionStrategyType` is `NotFound`? Shouldn't we throw a validation error?

Reply:
The execution strategy type of FileUpload strategy is actually NotFound, so I need to check for it there as well.